PYIC-8245: Add AIS stub

MikeCollingwood opened the pull request:
## Proposed changes

### What changed

- Add AIS stub

### Why did it change

- To allow us to integrate against a known interface and begin development/testing without dependency on the live AIS service.

### Issue tracking

- [PYIC-8245](https://govukverify.atlassian.net/browse/PYIC-8245)

## Checklists

<!-- Delete if changes in READMEs or documentation are not required -->
- [x] All READMEs and documentation updated where necessary

<!-- Delete if changes don't include risk of credentials being exposed -->
- [x] No risk of PII, credentials or anything else sensitive being exposed through logs

<!-- Delete if changes don't apply -->
- [x] Tests have been written/updated


[PYIC-8245]: https://govukverify.atlassian.net/browse/PYIC-8245?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

MikeCollingwood committed:
PYIC-8245: Add AIS stub

MikeCollingwood commented on di-ipv-ais-stub/openAPI/ais-external.yaml:
```diff
@@ -0,0 +1,368 @@
+          default:
+            statusCode: 200
+            responseTemplates:
+              application/json: '{"message": "Success!!"}'
```
New

MikeCollingwood commented on di-ipv-ais-stub/openAPI/ais-external.yaml:
```diff
@@ -0,0 +1,368 @@
+---
```
This is taken from AIS, there are only two additions:

blakeyp commented on .github/workflows/cri-ais-stub.yml:
```diff
@@ -0,0 +1,72 @@
+name: AIS CRI Stub Build Build & Push
```
Not a huge deal but it's not a CRI (also in the filename and README)

blakeyp commented on .github/workflows/cri-ais-stub.yml:
```diff
@@ -0,0 +1,72 @@
+      #  working-directory: ./update-ecs-env-vars/deploy
+      #  env:
+      #    ARTIFACT_BUCKET: ${{ secrets.UPDATE_ECS_ARTIFACT_BUCKET_NAME }}
+      #  run: sam package --s3-bucket="$ARTIFACT_BUCKET" --output-template-file=cf-template.yaml
```
bin this?

blakeyp commented on di-ipv-ais-stub/README.md:
```diff
@@ -0,0 +1,27 @@
+
+## Management Endpoint
+
+`POST /ais/{userId}` artificially primes a user with an account intervention response.
```
should be `/management/user/{userId}` ?

MikeCollingwood committed:
PYIC-8245: Fix documentation and OpenAPI spec

MikeCollingwood commented on di-ipv-ais-stub/openAPI/ais-external.yaml:
```diff
@@ -0,0 +1,367 @@
+          type: "string"
+          example: "error message"
+
+    AisManagementRequestBody:
+      description: Request body for AIS management
+      type: object
+      properties:
+        intervention:
+          type: "string"
+          example: "AIS_NO_INTERVENTION"
```
New

MikeCollingwood committed:
PYIC-8245: Fix README.md urls

MikeCollingwood committed:
PYIC-8245:
- Refactor cases into TS
- Fix entrypoint ref in CF
- Update core-dev-deploy template to work in dev
- Fix OpenAPI spec
- Update README
